create-container.sh

```diff
@@ -50,91 +58,108 @@ echo "  - Group: $DEVENV_GROUP"
 echo
 
 # Create container
-exist_container="$(sudo lxc-ls --filter ^"$NAME"$)"
-if [ -z "${exist_container}" ] ; then
-  echo "Creating container $NAME"
+EXIST_CONTAINTER="$(sudo lxc-ls --filter ^"$NAME"$)"
+if [ -z "${EXIST_CONTAINTER}" ] ; then
+  echo "Creating container $NAME..."
   sudo lxc-create --name "$NAME" -f "$LXC_CONFIG" -t download -l INFO -- --dist "$DISTRIBUTION" --release "$RELEASE" --arch "$ARCH"
 fi
-echo "Container ready"
+echo "Container is created"
 
 # Check if container is running, if not start it
-count=1
-while [ $count -lt $RETRIES ] && [ -z "$is_running" ]; do
-  is_running=$(sudo lxc-ls --running --filter ^"$NAME"$)
-  if [ -z "$is_running" ] ; then
-    echo "Starting container"
-    sudo lxc-start -n "$NAME" -d -l INFO
-    ((count++))
+COUNT=1
+IS_RUNNING=$(sudo lxc-ls --running --filter ^"$NAME"$)
+
+while [ -z "$IS_RUNNING" ]; do
+  # LOOP START#
+  # If container is not running after $RETRIES +1 attempts, then stop execution
+  if [ "$COUNT" -le "$RETRIES" ]; then
+    echo "Container not started, something is wrong."
+    echo "Please check log file /var/log/lxc/$NAME.log"
+    exit 1
   fi
+
+  ((COUNT++))
+  sleep 2
+
+  # LOOP BODY #
+  echo "Starting container..."
+  sudo lxc-start -n "$NAME" -d -l INFO
+
+  # POST END #
+  IS_RUNNING=$(sudo lxc-ls --running --filter ^"$NAME"$)
 done
 
-# If container is not running stop execution
-if [ -z "$is_running" ]; then
-  echo "Container not started, something is wrong."
-  echo "Please check log file /var/log/lxc/$NAME.log"
-  exit 0
-fi
-echo "Container is running..."
+echo "Container is running"
 
 # Wait to start container and check the IP
-count=1
-ip_container="$(sudo lxc-info -n "$NAME" -iH)"
-while [ $count -lt $RETRIES ] && [ -z "$ip_container" ] ; do
+COUNT=1
+IP_CONTAINER="$(sudo lxc-info -n "$NAME" -iH)"
+while [ -z "$IP_CONTAINER" ] ; do
+  # LOOP START #
+  if [ "$COUNT" -le "$RETRIES" ]; then
+    echo "Container is started but has no IP address."
+    echo "Please check log file /var/log/lxc/$NAME.log"
+    exit 1
+  fi
+
+  ((COUNT++))
+  echo "Waiting for container IP address..."
   sleep 2
-  echo "Waiting for container IP..."
-  ip_container="$(sudo lxc-info -n "$NAME" -iH)"
-  ((count++))
+
+  # LOOP END #
+  IP_CONTAINER="$(sudo lxc-info -n "$NAME" -iH)"
 done
-echo "Container IP: $ip_container"
+echo "Container has IP address: $IP_CONTAINER"
 echo
 
 # Add container IP to /etc/hosts
 echo "Removing old host $HOST from /etc/hosts"
 sudo sed -i '/'"$HOST"'/d' /etc/hosts
-host_entry="$ip_container       $HOST"
-echo "Add '$host_entry' to /etc/hosts"
-sudo -- sh -c "echo $host_entry >> /etc/hosts"
+HOST_ENTRY="\
+# LXC container for $NAME
+$IP_CONTAINER       $HOST"
+echo "Add '$HOST_ENTRY' to /etc/hosts"
```

Comment:
Substitute newline char so that $HOST_ENTRY is single line and the comment is also single line then.

```suggestion
echo "Add '${HOST_ENTRY/$'\n'/\\n}' to /etc/hosts"
```